server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3591,7 +3591,7 @@ public void testRecoverFromForeignTranslog() throws IOException {
             null,
             config.getRelativeTimeInNanosSupplier(),
             null,
-            false
+            true
```

Comment:
Are these new `true` I see in this PR a correction of an oversight from the old days? Or is it related to this PR?

Reply:
yeah I think that was an oversight, although maybe not an important one. In the public codebase all shards are promotable.